functests/ptp/ptp.go

```diff
@@ -108,4 +97,472 @@ var _ = Describe("ptp", func() {
+		if ds.Status.DesiredNumberScheduled == ds.Status.NumberReady {
+			return nil
+		}
+		timeout--
+		time.Sleep(time.Second)
+	}
+	return fmt.Errorf("Timeout")
+}
+
+func waitUntilLogIsDetected(pod v1core.Pod, timeout int, neededLog string) error {
+	var err error
+	var logs string
+	for timeout > 0 {
+		logs, err = getPodLogs(pod)
+		if strings.Contains(logs, neededLog) {
+			return nil
+		}
+		timeout--
+		time.Sleep(time.Second)
+	}
+	if err != nil {
+		return err
+	}
+	return fmt.Errorf("Log is not Found")
+}
+
+func getPtpPodOnNode(nodeName string) (v1core.Pod, error) {
+	runningPod, err := client.Client.Pods(ptpLinuxDaemonNamespace).List(metav1.ListOptions{LabelSelector: "app=linuxptp-daemon"})
+	Expect(err).NotTo(HaveOccurred(), fmt.Sprint("Error to get list of pods by label: app=linuxptp-daemon"))
+	Expect(len(runningPod.Items)).To(BeNumerically(">", 0), fmt.Sprint("PTP pods are  not deployed on cluster"))
+	for _, pod := range runningPod.Items {
+		if pod.Spec.NodeName == nodeName {
+			return pod, nil
+		}
+	}
+	return v1core.Pod{}, fmt.Errorf("Pod not found")
+}
+
+func replaceTestPod(pod v1core.Pod, timeout int) (v1core.Pod, error) {
+	err := client.Client.Pods(ptpLinuxDaemonNamespace).Delete(pod.Name, &metav1.DeleteOptions{
+		GracePeriodSeconds: pointer.Int64Ptr(0)})
+	Expect(err).NotTo(HaveOccurred())
+	for timeout > 0 {
+
+		newPod, podErr := getPtpPodOnNode(pod.Spec.NodeName)
+
+		if podErr == nil && newPod.Name != pod.Name && newPod.Status.Phase == "Running" {
+			return newPod, nil
+		}
+		timeout--
+		time.Sleep(time.Second)
+	}
+	return v1core.Pod{}, fmt.Errorf("Pod replace Error")
+}
+
+func getMasterSlaveAttachedInterfaces(pod v1core.Pod) []string {
+	var IntList []string
+	collectIntCommand := []string{"ls", "/sys/class/net/"}
+	commandOut, _ := execCommand(pod, collectIntCommand)
+	IntList = strings.Split(strings.Join(strings.Fields(commandOut), " "), " ")
+	return IntList
+}
+
+func getPtpMasterSlaveAttachedInterfaces(pod v1core.Pod) []string {
+	var ptpSupportedInterfaces []string
+	intList := getMasterSlaveAttachedInterfaces(pod)
+	for _, interf := range intList {
+		intParam := []string{"ethtool", "-T", interf}
+		_, buf := execCommand(pod, intParam)
+		if netParseEthtoolTimeStampFeature(&buf) {
+			ptpSupportedInterfaces = append(ptpSupportedInterfaces, interf)
+		}
+		time.Sleep(time.Second)
+	}
+	return ptpSupportedInterfaces
+}
+
+func getNonPtpMasterSlaveAttachedInterfaces(pod v1core.Pod) []string {
+	var ptpSupportedInterfaces []string
+
+	intList := getMasterSlaveAttachedInterfaces(pod)
+	for _, interf := range intList {
+		intParam := []string{"ethtool", "-T", interf}
+		_, buf := execCommand(pod, intParam)
+		if netParseEthtoolTimeStampFeature(&buf) == false {
+			ptpSupportedInterfaces = append(ptpSupportedInterfaces, interf)
+		}
+		time.Sleep(time.Second)
+	}
+	return ptpSupportedInterfaces
+}
+
+func elementPartOfList(element string, list []string) bool {
```

Comment:
there is a build-in function in ginko for this. you can remove this function and use `ContainElement`